sdv/constraints/tabular.py

```diff
@@ -211,78 +214,109 @@ class GreaterThan(Constraint):
         drop (str):
             Which column to drop during transformation. Can be ``'high'``,
             ``'low'`` or ``None``.
-        high_is_scalar(bool or None):
-            Whether or not the value for high is a scalar or a column name.
-            If ``None``, this will be determined during the ``fit`` method
-            by checking if the value provided is a column name.
-        low_is_scalar(bool or None):
-            Whether or not the value for low is a scalar or a column name.
-            If ``None``, this will be determined during the ``fit`` method
-            by checking if the value provided is a column name.
+        scalar (str):
+            Which value is a scalar. Can be ``'high'``, ``'low'`` or ``None``.
+            If ``None`` then both ``high`` and ``low`` are column names.
     """
 
-    _diff_column = None
+    _diff_columns = None
     _is_datetime = None
-    _column_to_reconstruct = None
-
-    def __init__(self, low, high, strict=False, handling_strategy='transform',
-                 fit_columns_model=True, drop=None, high_is_scalar=None,
-                 low_is_scalar=None):
-        self._low = low
-        self._high = high
-        self._strict = strict
-        self.constraint_columns = (low, high)
-        self._drop = drop
-        self._high_is_scalar = high_is_scalar
-        self._low_is_scalar = low_is_scalar
-        super().__init__(handling_strategy=handling_strategy,
-                         fit_columns_model=fit_columns_model)
-
-    def _get_low_value(self, table_data):
-        if self._low_is_scalar:
-            return self._low
-        elif self._low in table_data.columns:
-            return table_data[self._low]
+    _columns_to_reconstruct = None
 
-        return None
+    @staticmethod
+    def _validate_scalar(scalar_column, column_names, scalar):
+        """Validate scalar comparison inputs.
 
-    def _get_high_value(self, table_data):
-        if self._high_is_scalar:
-            return self._high
-        elif self._high in table_data.columns:
-            return table_data[self._high]
+        - Make sure that the scalar column is not a list and raise the proper error if it is.
+        - If the `column_names` is not a list it would make it a list.
+        - Return both the scalar column and column names with the right format
+        """
+        if isinstance(scalar_column, list):
+            raise TypeError(f'{scalar} cannot be a list when scalar="{scalar}".')
+
+        column_names = [column_names] if not isinstance(column_names, list) else column_names
```

Comment:
I would rather do this in a 2 lines if. Otherwise we are just reassigning unnecessarily:

```
if not isinstance(column_names, list):
     column_names = [column_names]
```